### PR TITLE
New example: Dialin bot (call your Pipecat via phone)

### PR DESCRIPTION
**Pull request by jptaylor**
- [x] Update Daily bot with latest REST URL
- [ ] Link to docs page once merged (post example merge)
- [ ] Create tests  (post example merge)

### CODE REVIEW

File: examples/dialin-chatbot/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<div align="center">
+ <img alt="pipecat" width="300px" height="auto" src="image.png">
+</div>
+
+# Dialin example
+
+Example project that demonstrates how to add phone number dialin to your Pipecat bots. We include examples for both Daily (`bot_daily.py`) and Twilio (`bot_twilio.py`), depending on who you want to use as a phone vendor.
+
+- 🔁 Transport: Daily WebRTC
+- 💬 Speech-to-Text (transcription): Daily
```
Comment (kwindla) [inline examples/dialin-chatbot/README.md:10]: I think it's good to credit Deepgram here. Maybe "Deepgram via Daily transport". wdyt?

File: examples/dialin-chatbot/bot_daily.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            call_domain=callDomain,
+            sip_uri=sipUri
+        )
+
+        transport = DailyTransport(
+            room_url,
+            token,
+            "Chatbot",
+            DailyParams(
+                api_url=f"https://{daily_api_url}",
```
Comment (kwindla) [inline examples/dialin-chatbot/bot_daily.py:47]: Is there a reason to hardcode the `https` here in the code?

File: examples/dialin-chatbot/daily_helpers.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    def __init__(self, daily_api_key: str, daily_api_url: str = "api.daily.co/v1"):
+        self.daily_api_key = daily_api_key
+        self.daily_api_url = daily_api_url
+
+    def _get_name_from_url(self, room_url: str) -> str:
+        return urllib.parse.urlparse(room_url).path[1:]
+
+    def create_room(self, params: DailyRoomParams) -> DailyRoomObject:
+        res = requests.post(
+            f"https://{self.daily_api_url}/rooms",
```
Comment (kwindla) [inline examples/dialin-chatbot/daily_helpers.py:64]: Same question about https.

File: examples/dialin-chatbot/daily_helpers.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        try:
+            room = DailyRoomObject(**data)
+        except ValidationError as e:
+            raise Exception(f"Invalid response: {e}")
+
+        return room
+
+    def _get_room_from_name(self, room_name: str) -> DailyRoomObject:
+        res: requests.Response = requests.get(
+            f"https://{self.daily_api_url}/rooms/{room_name}",
```
Comment (kwindla) [inline examples/dialin-chatbot/daily_helpers.py:83]: Same question about https.

File: examples/dialin-chatbot/daily_helpers.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        if not room_url:
+            raise Exception(
+                "No Daily room specified. You must specify a Daily room in order a token to be generated.")
+
+        expiration: float = time() + expiry_time
+
+        room_name = self._get_name_from_url(room_url)
+
+        res: requests.Response = requests.post(
+            f"https://{self.daily_api_url}/meeting-tokens",
```
Comment (kwindla) [inline examples/dialin-chatbot/daily_helpers.py:113]: Same question about https.

File: src/pipecat/transports/services/helpers/daily_rest.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    def __init__(self, daily_api_key: str, daily_api_url: str = "api.daily.co/v1"):
+        self.daily_api_key = daily_api_key
+        self.daily_api_url = daily_api_url
+
+    def _get_name_from_url(self, room_url: str) -> str:
+        return urllib.parse.urlparse(room_url).path[1:]
+
+    def create_room(self, params: DailyRoomParams) -> DailyRoomObject:
+        res = requests.post(
+            f"https://{self.daily_api_url}/rooms",
```
Comment (kwindla) [inline src/pipecat/transports/services/helpers/daily_rest.py:69]: Same question about https.

File: src/pipecat/transports/services/helpers/daily_rest.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+        try:
+            room = DailyRoomObject(**data)
+        except ValidationError as e:
+            raise Exception(f"Invalid response: {e}")
+
+        return room
+
+    def _get_room_from_name(self, room_name: str) -> DailyRoomObject:
+        res: requests.Response = requests.get(
+            f"https://{self.daily_api_url}/rooms/{room_name}",
```
Comment (kwindla) [inline src/pipecat/transports/services/helpers/daily_rest.py:88]: Same question about https.

File: src/pipecat/transports/services/helpers/daily_rest.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+        if not room_url:
+            raise Exception(
+                "No Daily room specified. You must specify a Daily room in order a token to be generated.")
+
+        expiration: float = time() + expiry_time
+
+        room_name = self._get_name_from_url(room_url)
+
+        res: requests.Response = requests.post(
+            f"https://{self.daily_api_url}/meeting-tokens",
```
Comment (kwindla) [inline src/pipecat/transports/services/helpers/daily_rest.py:118]: Same question about https.

**Review by vr000m — APPROVED** [review]
LGTM, tested both pipeline on production on Thursday and Friday.

**Review by vr000m — APPROVED** [review]
LGTM

**Review by kwindla — APPROVED** [review]
Lgtm

**Pull request merged**